tests/integration/test_permissions.py

```diff
@@ -361,47 +162,64 @@ def test_inspire_search_filter(app, app_client, user_info, total_count, es_filte
     assert LiteratureSearch().count() == total_count
 
 
+@pytest.mark.parametrize('user_info,total_count,es_filter', [
+    # anonymous user
+    (None, 0, [{'bool': {'must_not': [{'match': {'_collections': u'HERMES Internal Notes'}}], 'must': [{'match': {'_collections': u'HERMES Internal Notes'}}]}}]),
+    (dict(email='johndoe@inspirehep.net', password='123456'), 0, [{'bool': {'must_not': [{'match': {'_collections': u'HERMES Internal Notes'}}], 'must': [{'match': {'_collections': u'HERMES Internal Notes'}}]}}]),
+    (dict(email='cataloger@inspirehep.net', password='123456'), 0, [{'bool': {'must_not': [{'match': {'_collections': u'HERMES Internal Notes'}}], 'must': [{'match': {'_collections': u'HERMES Internal Notes'}}]}}]),
+    (dict(email='hermescataloger@inspirehep.net', password='123456'), 1, [{'match': {'_collections': 'HERMES Internal Notes'}}]),
+    (dict(email='admin@inspirehep.net', password='123456'), 1, [{'match': {'_collections': 'HERMES Internal Notes'}}]),
+])
+def test_inspire_search_filter_restricted_collection(app, app_client, user_info, total_count, es_filter):
+    """Test default inspire search filter."""
+
+    if user_info:
+        login_user_via_view(app_client, email=user_info['email'],
+                            password=user_info['password'],
+                            login_url='/login/?local=1')
+
+    # Doing a client request creates a request context that allows the
+    # assert to correctly use the logged in user.
```

Comment:
Better: https://github.com/inspirehep/inspire-next/blob/fcb1263266d851ba4651a0123455ad2ff3974916/tests/unit/conftest.py#L75-L78

Reply:
(You'll need to copy this from `tests/unit/conftest.py` to `tests/integration/conftest.py`.)

Reply:
Tried this but does not work, at least just with `with app.test_request_context()`